internal/locate/region_cache.go

```diff
@@ -671,6 +672,12 @@ func NewRegionCache(pdClient pd.Client) *RegionCache {
 	c.notifyCheckCh = make(chan struct{}, 1)
 	c.bg = newBackgroundRunner(context.Background())
 	c.enableForwarding = config.GetGlobalConfig().EnableForwarding
+	if c.pdClient != nil {
+		c.clusterID = c.pdClient.GetClusterID(context.Background())
+	}
+	if c.clusterID == 0 {
+		logutil.BgLogger().Error("cluster id is not set properly")
+	}
```

Comment:
`RegionCache` exports a method `SetPDClient`, maybe we can move the code of cluster id initialization there.

Reply:
The `SetPDClient` is marked as `for test only` in the comments.